### ISSUE TITLE
Use einsum on ADF calculator

### ISSUE THREAD

**Issue by SamTov** [zincware/MDSuite#245]
**What feature would you like to see added?**
Currently it is not possible for us to always use einsum in the ADF computation. Rather we allow the user to select tf.function which will use considerable more memory but is much faster. Einsum would likely improve the speed further. Once we have access to more intelligent memory management, we should allow for internal selection of the einsum procedure to optimized computation time.

**Describe how this feature would improve the code**
It would make it much faster.

**Describe how this feature would work/possible implementations**
We would need to check that the machine can handle the memory load and if so, use einsum. We must keep in mind that some monitoring may be necessary to ensure other processes don't disrupt it, hence the necessity for intelligent memory management.

**Comment by PythonFZ** [comment]
related to #248

**Comment by SamTov** [comment]
Is this still relevant?

**Comment by PythonFZ** [comment]
Yes - the einsum calculation would be superior to the current one. The downside is the memory usage which has not been worked out yet.

**Comment by SamTov** [comment]
Right so ideally better memory management would mean this could be implemented?

**Comment by PythonFZ** [comment]
It would also require some rewriting of the ADF code - but that might not be a bad thing anyway.

**Comment by SamTov** [comment]
Yes. With the restructure I did a few days ago it should be a little easier to identify restructure parts as it is fairly modular now
